src/client.rs

```diff
@@ -1142,14 +1146,19 @@ impl Client {
         image: &Reference,
         layer: impl AsLayerDescriptor,
         offset: u64,
+        length: Option<u64>,
     ) -> Result<BlobResponse> {
-        let response = self.pull_blob_response(image, &layer, Some(offset)).await?;
+        let response = self
+            .pull_blob_response(image, &layer, Some(offset), length)
+            .await?;
 
         let status = response.status();
         match status {
-            StatusCode::OK => Ok(BlobResponse::Full(stream_from_response(response, &layer)?)),
+            StatusCode::OK => Ok(BlobResponse::Full(stream_from_response(
+                response, &layer, true,
+            )?)),
             StatusCode::PARTIAL_CONTENT => Ok(BlobResponse::Partial(stream_from_response(
-                response, &layer,
+                response, &layer, false,
```

Comment:
If the `offset` was ever non-zero this method should already have been failing on stream verification? I checked registry spec and couldn't see anything about returned specialized digests for partial reads.

Reply:
It makes sense that partial pulls can't be validated, so we should probably document the `pull_blob_stream_partial` that partial pulls by nature cannot be validated and as such should be validated by the caller when complete

Reply:
Actually, now that I'm looking at this more, we might need to return the two possible digests to validate against to the caller on a partial request, otherwise they won't have access to the digest of the header (probably as new data members of `SizedStream`). What do other clients do/expect to handle digest validation in these cases?

Reply:
@thundergolfer If you'd like/are willing to do this, that would be great! If not, let me know and I'll switch this to an approval and I'll change it after merge since it is a preexisting edge case we didn't account for originally

Reply:
Hey @thomastaylor312, I've switched focus off this project so I'll be unlikely to get to it soon.

Reply:
Ok sounds good. I'll merge this as is and fix up. Thanks for adding this!